src/ConstructiveSolidGeometry/LinePrimitives/Line.jl

```diff
@@ -1,8 +1,29 @@
-struct Line{T} <: AbstractLinePrimitive{T}
+struct Line{T<:AbstractFloat} <: AbstractLinePrimitive{T}
```

Comment:
`<:AbstractFloat` is already in `abstract type AbstractGeometry{T <: AbstractFloat} end`.
Please remove it here.

Reply:
That is true, but when calling Line() the compiler will skip the type promotion and call directly the struct. This leads to an error, since now Int is handed to the struct.
If I explicitly state T<:AbstractFloat, Line() will call the type promotion function